src/test/java/com/google/gcs/sdrs/service/worker/rule/impl/StsRuleExecutorTest.java

```diff
@@ -86,6 +102,64 @@ public void globalRuleExecutionWithDatasetType() {
     }
   }
 
+  @Test
+  public void executeDatasetRuleSuccess() throws IOException {
+    Collection<RetentionRule> datasetRules = new HashSet<>();
+    datasetRules.add(testRule);
+
+    TransferJob transferJob = createBasicTransferJob();
+
+    when(PrefixGeneratorUtility.generateTimePrefixes(
+        any(), any(), (ZonedDateTime) notNull())).thenCallRealMethod();
```

Comment:
The unit test failed when I tried. I think it's caused by thenCallRealMethod(). In general, methods are mocked in unit tests. A real method call has too many dependency. From the stacktrace below, it seems that it calls Pub/Sub API. It worked in your environment you probably had the right ADC set up. However, unit tests are run without the setup. 

java.lang.RuntimeException: javax.net.ssl.SSLException: unable to setup trustmanager

	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.buildTransportFactory(NettyChannelBuilder.java:379)
	at io.grpc.internal.AbstractManagedChannelImplBuilder.build(AbstractManagedChannelImplBuilder.java:406)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createSingleChannel(InstantiatingGrpcChannelProvider.java:246)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createChannel(InstantiatingGrpcChannelProvider.java:165)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.getTransportChannel(InstantiatingGrpcChannelProvider.java:152)
	at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:149)
	at com.google.cloud.pubsub.v1.stub.GrpcPublisherStub.create(GrpcPublisherStub.java:161)
	at com.google.cloud.pubsub.v1.Publisher.<init>(Publisher.java:149)
	at com.google.cloud.pubsub.v1.Publisher.<init>(Publisher.java:78)
	at com.google.cloud.pubsub.v1.Publisher$Builder.build(Publisher.java:614)
	at com.google.gcs.sdrs.service.mq.PubSubMessageQueueManagerImpl.getInstance(PubSubMessageQueueManagerImpl.java:66)
	at com.google.gcs.sdrs.service.worker.rule.impl.StsRuleExecutor.sendInactiveDatasetNotification(StsRuleExecutor.java:876)

Reply:
I think this is caused by https://github.com/GoogleCloudPlatform/storage-sdrs/blob/2b2933dff7f1247393277c78199d42e42b4322e1/src/main/java/com/google/gcs/sdrs/service/worker/rule/impl/StsRuleExecutor.java#L221

I've mocked this in the unit tests also. I think it will work.